SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityNewTaskChecker.java

```diff
@@ -368,16 +382,27 @@ CheckTaskState getTaskState(SingularityTask task, Optional<SingularityRequestWit
     return Optional.absent();
   }
 
-  private boolean isOverdue(SingularityTask task) {
-    final long taskDuration = System.currentTimeMillis() - task.getTaskId().getStartedAt();
+  private boolean isHealthcheckOverdue(SingularityTask task) {
+    final long healthcheckDuration = taskManager.getLastHealthcheck(task.getTaskId()).get().getDurationMillis().or(0L);
```

Comment:
in task history updates, there will be an update with status of `TASK_RUNNING` that has an associated timestamp